lib/src/test/java/growthbook/sdk/java/GBFeaturesRepositoryTest.java

```diff
@@ -137,8 +139,65 @@ void canFetchEncryptedFeatures_mockedResponse() throws IOException, FeatureFetch
+
+        String expected = "{\"banner_text\":{\"defaultValue\":\"Welcome to Acme Donuts!\",\"rules\":[{\"condition\":{\"country\":\"france\"},\"force\":\"Bienvenue au Beignets Acme !\"},{\"condition\":{\"country\":\"spain\"},\"force\":\"¡Bienvenidos y bienvenidas a Donas Acme!\"}]},\"dark_mode\":{\"defaultValue\":false,\"rules\":[{\"condition\":{\"loggedIn\":true},\"force\":true,\"coverage\":0.5,\"hashAttribute\":\"id\"}]},\"donut_price\":{\"defaultValue\":2.5,\"rules\":[{\"condition\":{\"employee\":true},\"force\":0}]},\"meal_overrides_gluten_free\":{\"defaultValue\":{\"meal_type\":\"standard\",\"dessert\":\"Strawberry Cheesecake\"},\"rules\":[{\"condition\":{\"dietaryRestrictions\":{\"$elemMatch\":{\"$eq\":\"gluten_free\"}}},\"force\":{\"meal_type\":\"gf\",\"dessert\":\"French Vanilla Ice Cream\"}}]}}";
+
+        subject.onFeaturesRefresh(new FeatureRefreshCallback() {
+            @Override
+            public void onRefresh(String featuresJson) {
+                assertEquals(expected, featuresJson);
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+            }
+        });
+
+        subject.getFeaturesJson();
+    }
+
+    @Test
+    void refreshCallbackWhenCacheRefreshFailed() throws IOException, FeatureFetchException, InterruptedException {
+        String fakeResponseJson = "{\"status\":200,\"features\":{\"banner_text\":{\"defaultValue\":\"Welcome to Acme Donuts!\",\"rules\":[{\"condition\":{\"country\":\"france\"},\"force\":\"Bienvenue au Beignets Acme !\"},{\"condition\":{\"country\":\"spain\"},\"force\":\"¡Bienvenidos y bienvenidas a Donas Acme!\"}]},\"dark_mode\":{\"defaultValue\":false,\"rules\":[{\"condition\":{\"loggedIn\":true},\"force\":true,\"coverage\":0.5,\"hashAttribute\":\"id\"}]},\"donut_price\":{\"defaultValue\":2.5,\"rules\":[{\"condition\":{\"employee\":true},\"force\":0}]},\"meal_overrides_gluten_free\":{\"defaultValue\":{\"meal_type\":\"standard\",\"dessert\":\"Strawberry Cheesecake\"},\"rules\":[{\"condition\":{\"dietaryRestrictions\":{\"$elemMatch\":{\"$eq\":\"gluten_free\"}}},\"force\":{\"meal_type\":\"gf\",\"dessert\":\"French Vanilla Ice Cream\"}}]}},\"dateUpdated\":\"2023-01-11T00:26:01.745Z\"}";
+        OkHttpClient okHttpClient = mockHttpClientFailedCallback(fakeResponseJson, new IOException("Request failed"));
+        int ttlSeconds = 0;
+
+        GBFeaturesRepository subject = new GBFeaturesRepository(
+                okHttpClient,
+                "http://localhost:80",
+                null,
+                ttlSeconds
+        );
+        subject.initialize();
 
+        subject.onFeaturesRefresh(new FeatureRefreshCallback() {
+            @Override
+            public void onRefresh(String featuresJson) {
+            }
 
+            @Override
+            public void onError(Throwable throwable) {
+                assertThrows(IOException.class, () -> {
+                    throw throwable;
+                });
+            }
+        });
```

Comment:
I don't think this assertion is working as expected. I'm assuming you're trying to test that the onError callback is called when it fails, but when I comment out the real code, this test still passes, which leads me to believe it isn't testing anything.

Reply:
I slightly corrected the test, if I understood you correctly, now when you comment on the real code, it ends with an error. If you have any ideas how it can be implemented, I will be grateful)

Reply:
There is some strange behaviour with the tests still. I notice that the `getResponse` method you've abstracted from the initial success is used for the failure, and it responds with a 200. I see that when I change the failed response JSON, it fails because it tries to successfully process the failed JSON response data. It looks like both the success and failure cases are being called in the tests. I don't think the failure tests are accurately being tested—we are testing a condition that would never occur (both a success and a failure). You can see the odd behaviour by changing `testOnFeaturesRefresh_Error()`'s fake response to be `"{}"`—we get: `CONFIGURATION_ERROR : No features found` which means it's trying to run the success.

Reply:
oh yeah i see...it's happening because i'm calling the subject.initialize() method, it's not really related to the error check as it doesn't call onRefreshFailed, i can remove it from the call and leave only subject.getFeaturesJson(), then the test will work correctly